### PR TITLE
Helper struct for ser/deserializing SignerSetV2s in a more human-friendly fashion

### PR DESCRIPTION
**Pull request by eranrund**
### Motivation

`SignerSetV2`s are going to be stored inside JSON configuration files
suck as `tokens.json` or files used when constructing `MintConfigTx`s
(once the mint client starts supporting the revised multisig scheme).

The current representation of `SignerSetV1` inside JSON files has caused
some headaches:
1. It is not possible to tell which entity stands behind which public
   key. Separately, it would be nice to refer to a public key by name when declaring a signer set.
2. There is currently a known bug in `pem::parse_many` - turns out it
   will silently ignore data that is not properly surrounded by the
dashed markers. The format being proposed here parses each PEM
individually, so it bypasses this annoyance.

Basically the idea is to create a slightly more human-friendly
serialized representation of signer sets.

### CODE REVIEW

**Comment by eranrund** [comment]
Current dependencies on/for this PR:
* master
  * **PR #2657** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2657" target="_blank">Graphite</a> 
    * **PR #2661** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2661" target="_blank">Graphite</a> 
      * **PR #2671** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2671" target="_blank">Graphite</a> 
        * **PR #2699** <a href="https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2699" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/mobilecoin/2699?utm_source=stack-comment).

**Comment by eranrund** [comment]
Closing in favor of #2732 and the chain of PRs that follow it. That approach turned out to result in significantly less changes.
